### PR TITLE
change inspect to return receiver / add inspector

### PR DESCRIPTION
**Pull request by SabineMa**
https://github.com/pharo-project/pharo/issues/11803

### CODE REVIEW

**Review by jordanmontt — APPROVED** [review]

**Comment by Ducasse** [comment]
Would be good to add a test to cover that we do not get this regression.

**Comment by SabineMa** [comment]
I can add

testInspector
self assert: 1 inspector class equals: SpWindowPresenter

and

testInspect
self assert: 1 inspect equals: 1


Is this ok?
I am not sure to which class.
what is the condition that the regression will not occur?

File: src/Tool-Base/Object.extension.st [hunk 1]
```diff
@@ -18,9 +18,19 @@ Object >> browseHierarchy [
...
+	^ self
+]
+
+{ #category : #'*Tool-Base' }
+Object >> inspector [
+	"Create and schedule an Inspector in which the user can examine the receiver's variables.
+	Return the inspector window."
 
-	^ Smalltalk tools inspector inspect: self
+	^ Smalltalk tools inspector inspect: self.
```
Comment (astares) [inline src/Tool-Base/Object.extension.st:33]: Please no unnecessary dots at the end.

**Review by astares — CHANGES_REQUESTED** [review]

**Comment by jordanmontt** [comment]
Also maybe you could add a comment in the methods as a form of documentation. For example, in the `inspect`method saying that if you can the window object you should use `inspector`

**Comment by Ducasse** [comment]
Hi sabina 
Yes this sounds good to me and they should be packaged with the same tools-tests or something like that.

**Comment by SabineMa** [comment]
https://github.com/pharo-project/pharo/issues/11803#issuecomment-1283836624
Please do not integrate

**Comment by SabineMa** [comment]
Everything is ok as it is. I was confused.

**Review by jordanmontt — COMMENTED** [review]
As discussed in the issue, I also think we should add:

- deprecate basicInspectand renamed it to basicInspector. We still return the spec window.
- refactor inspectOnce to always return self.

**Comment by SabineMa** [comment]
I already wrote the code but I have auth problems for doing the pull request. 
I have to manage other stuff with you product and will come back here asap.

**Comment by jordanmontt** [comment]
Hello @SabineMa , did you have the time to commit your changes?
If not, tomorrow there is a Pharo sprint, this can be an issue for that :)

**Comment by SabineMa** [comment]
closed because done here: https://github.com/pharo-project/pharo/pull/11886

**Pull request closed without merging**